deps/juggler-v3/test.js

```diff
@@ -1,10 +1,48 @@
-// Copyright IBM Corp. 2015,2019. All Rights Reserved.
+// Copyright IBM Corp. 2019. All Rights Reserved.
 // Node module: loopback-connector-cloudant
 // This file is licensed under the Artistic License 2.0.
 // License text available at https://opensource.org/licenses/Artistic-2.0
 
 'use strict';
 
+const should = require('should');
+const juggler = require('loopback-datasource-juggler');
+const name = require('./package.json').name;
+
+require('../../test/init');
+
+
+
+/**  from couldant.connection.test  */
+describe('cloudant connection', function() {
+    context('with an invalid cloudant connection', function() {
+      it('returns error with fake url', function(done) {
+        var fakeConfig = {
+          url: 'http://fake:foo@localhost:4',
+        };
+        var fakeDB = global.getDataSource(fakeConfig);
+        fakeDB.once('error', function(err) {
+          should.exist(err);
+          err.message.should.match(/error happened in your connection/);
+          done();
+        });
+      });
+    });
+  });
+/**  from couldant.connection.test  */
+
+
+
+describe(name, function() {
+  before(function() {
+    return global.resetDataSourceClass(juggler.DataSource);
+  });
+
+  after(function() {
+    return global.resetDataSourceClass();
```

Comment:
To make this work, you need to modify `test/init.js`, add `resetDataSourceClass` helper and modify `getDataSource` to allow configuration of the data-source class used. See https://github.com/strongloop/loopback-connector-mongodb/pull/519/files#diff-40dfb1c53004f1488cbe6eb300b286dd